src/Workspaces/Core/Portable/Workspace/Solution/Solution.cs

```diff
@@ -458,13 +497,8 @@ internal Solution WithRunAnalyzers(ProjectId projectId, bool runAnalyzers)
         /// </summary>
         public Solution AddProjectReference(ProjectId projectId, ProjectReference projectReference)
         {
-            var newState = _state.AddProjectReferences(projectId, SpecializedCollections.SingletonEnumerable(projectReference));
-            if (newState == _state)
-            {
-                return this;
-            }
-
-            return new Solution(newState);
+            return AddProjectReferences(projectId, SpecializedCollections.SingletonEnumerable(
+                projectReference ?? throw new ArgumentNullException(nameof(projectReference))));
```

Comment:
The indenting concern was the => but that it looks like "projectReference ?? ..." is an argument to AddProjectReferences but it's *not* since there's extra wrapping there. I think we'd usually start the "SpecializedCollections.SingletonEnumerable" on it's own line and you end up with three.